code/modules/reagents/reagent_containers/borghydro.dm

```diff
@@ -252,3 +253,98 @@ Borg Shaker
 	desc = "An advanced chemical synthesizer and injection system, designed to stabilize patients."
 	reagent_ids = list(/datum/reagent/medicine/epinephrine)
 	accepts_reagent_upgrades = FALSE
+
+//////////////////////
+//borg beaker holder//
+//////////////////////
+
+/**An arm that only holds beakers. Can drop them, place them into machines, or onto tables. Attacks made by the arm, or
+to the arm are passed onto a stored beaker, if one exists. */
+
+/obj/item/borg_beaker_holder
+	name = "beaker storage apparatus"
+	desc = "A special apparatus for carrying beakers without spilling the contents."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "borg_beaker_apparatus"
+	var/obj/item/reagent_containers/stored
+	var/mob/living/silicon/robot/robot
+
+/obj/item/borg_beaker_holder/Initialize()
+	var/obj/item/robot_module/module = loc
+	robot = module.loc
+	stored = new /obj/item/reagent_containers/glass/beaker/large
+	stored.forceMove(src)
+	update_icon()
+	. = ..()
+
+/obj/item/borg_beaker_holder/examine()
+	. = ..()
+	if(stored)
+		. += "The apparatus currently has [stored] secured, which contains:"
+		if(length(stored.reagents.reagent_list))
+			for(var/datum/reagent/R in stored.reagents.reagent_list)
+				. += "[R.volume] units of [R.name]"
+		else
+			. += "Nothing."

```

Comment:
Oh, also mention alt-click behaviour in the examine por favor.

Reply:
It actually doesn't work by alt-clicking. It seems clicking on a module doesn't use the attack_self like it would if a human did it. But I've added the Alt-Z note to the desc, as well as a note about right-clicking to use the context menu drop beaker verb.